spring-kafka/src/main/java/org/springframework/kafka/listener/ConcurrentMessageListenerContainer.java

```diff
@@ -355,6 +355,7 @@ protected void doStop(final Runnable callback, boolean normal) {
 				}
 			}
 			for (KafkaMessageListenerContainer<K, V> container : this.containers) {
+				container.setAllowedToStart(false);
```

Comment:
I would really be more specific here and call it `setOrphaned()` (or `fenced`) as you call it in the exception message.